### DISCUSSION TITLE
Incorrect Video Date Display

### DISCUSSION THREAD

**Discussion by InstantRamenNoodles** [deckerst/aves#928] [Q&A]
I just started using AVES due to issues with proper display of video metadata with Google Gallery. It works almost perfectly, but I have 1 issue: AVES seems to correctly pull the dates from the exif data, however the time is all off by a factor of 4 or 5 hours. After messing with the exif data of a few test files, I noticed that the time zone data is all -04:00 or -05:00 (obvious since I am in the Eastern Time Zone). The date and time when viewed in exiftool are displayed as, for example January 21st 2021 at 9 pm, 2021:01:21 21:00:00-05:00. AVES would display such a video as having a date of 16:00:00, with all the remaining info being correct.

From what I can see, its as if AVES is treating the time zone data as a subtraction operation rather than an add-on with a dash connection the two. Is there anything I can do about this? I kind of don't want to live with the fact that all my videos are off by a factor of 4 or 5 hours.

For future reference, since I can't seem to find the information I'd also like to ask which date in the video metadata AVES uses to sort them since I can't seem to find this information. I apologize if it's somewhere obvious and I just missed it.

Any help is appreciated!

**Comment by deckerst** [comment]
There's a lot going on.

About Exif.
There's no Exif in video files. In videos, there can be format specific metadata and XMP. Exif can be embedded in certain photo formats. Exif date fields do not include the time zone. Time zone fields do exist in Exif, but they are often missing.

About video dates.
Regarding the minus sign misinterpretation, it's possible but then you'd likely end up with a 8~10 hour difference with the truth (using +5 instead of -5). If you say the difference is the same as your timezone offset, I guess the timezone is simply ignored. If you could provide a sample video, I'd like to take a closer look.

> For future reference, since I can't seem to find the information I'd also like to ask which date in the video metadata AVES uses to sort them since I can't seem to find this information. I apologize if it's somewhere obvious and I just missed it.

I need to take a look, but often dates presented by Aves are composite, with fallbacks, depending on the format, library support, which fields are present, etc.

**Reply by InstantRamenNoodles** [reply]
Hello,

I appreciate the clarification. Also, for the record the dates are written into the metadata in Eastern time and then the Timezone information is given in addition. I have attached an example video. The correct time is 6:55 pm EST, and it lists the time zone as -05:00. However, Aves displays it as 1:55 pm. I've also added an example of a video that is displayed correctly, to see if there is potentially a difference in how the metadata is written.

One other thing, it doesn't seem to be all my videos that have this issue. A lot of the videos were taken on an older phone and transferred over, and the only thing that seemed to be a difference the good and bad videos is whether they were taken/created on that phone or not. Not sure why this would be the case as I manually edited the metadata of all my videos in bulk, but I can't find any other difference that seems notable. I also don't know if every video taken on that phone is messed up, but I haven't seen a messed up video that was taken on that phone so far.

I'd really appreciate any insight you could give. I'm kind of hitting a wall.

**Comment by deckerst** [comment]
Thanks for the sample.

Just so you know, your video taken with an iPhone 11 includes GPS coordinates (which are not recognized by Aves, by the way >.<), so you may want to remove the links if you care about such things.

And your cats are delightful.

**Comment by deckerst** [comment]
Here are all the dates in your 2 samples:
<img width="250" src="https://github.com/deckerst/aves/assets/13898333/599d2cb9-d451-4f16-be09-92e6be06d974"/><img width="250" src="https://github.com/deckerst/aves/assets/13898333/cf4be11e-f49e-408e-9593-9c480469beae"/>

The `QuickTime Metadata` section shows a date with the right time zone. However, all the others show times specifying it as UTC (the "Z" at the end of "2019-12-10T18:55:56Z" stands for "Zulu Time Zone", i.e. no offset), or not specifying a time zone at all (like the Exif one inside the `XMP` section).

By the way, I wrote earlier that there is no Exif in videos, but here it is. What I meant is that there is no native Exif section planned by the video formats, but XMP allows embedding whatever so it can embed Exif fields too. Personally I think it's cramming a square through a circle ^^

**Reply by InstantRamenNoodles** [reply]
Hi,

I think that basically solves it. I did realize some of the metadata streams wouldn't accept time zone data, but I didn't realize that meant they were being processed as GMT time. All I have to do now is modify the metadata of all my videos to follow the same standard of GMT time in those fields and correctly modified time zone in the remaining fields, and it should probably work from there.

I'll also keep in mind what you said about video metadata. I used a generalized command with wildcards over all file extensions assuming they all had similar fields. I'll probably leave it as is since I don't think it will cause issues in the future (since if it's not natively supported I can't imagine any software would attempt to read it), but in the future I'll keep it in mind when writing to video files.

Thanks a lot for your response! I can figure out the rest from here.
